### ISSUE TITLE
Wrong cast error in `assoc-in`

### ISSUE THREAD

**Issue by helins** [Convex-Dev/convex#384]
Gen tests found this one:

```clojure
(assoc-in (quote {false #797,(39) {},0x5f \b,() false}) (quote (() #0)) (quote #0))
```
```
Exception code:
  :CAST
Message:
  Can't convert value of type Nil to type DataStructure
Trace:
  In function: assoc-in
```

Where does that `nil` come from?

**Comment by mikera** [comment]
It's presumably a nil that you get from looking up a non-existent value in the map?

Better error message might be possible.

**Comment by helins** [comment]
But the first item in the path is `()` which should point to `false`, right?

**Comment by mikera** [comment]
Ah got it I think I have a fix! Will be in 0.7.8
